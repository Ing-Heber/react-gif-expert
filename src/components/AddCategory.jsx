import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form aria-label="form" onSubmit={onSubmit}>
            <input type="text" placeholder="Search Gif" value={inputValue} onChange={onInputChange} />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
