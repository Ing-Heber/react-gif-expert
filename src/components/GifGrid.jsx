import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    // Never call functions inside funcional components
    // getGifs(category);

    return (
        <>
            <h3>{category}</h3>

            {isLoading && (<h2>Loading...</h2>)}

            <div className='card-grid'>
                {/* {images.map(({ id, title, url }) => <GifItem key={id} title={title} url={url}  /> )} */}
                {images.map((image) => <GifItem key={image.id} {...image} />)}
            </div>

        </>
    )

};