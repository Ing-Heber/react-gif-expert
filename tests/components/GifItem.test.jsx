import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem';

describe('Testing <GifItem /> Component',
    () => {

        const testTitle = 'One Punch';
        const testUrl = 'https://localhost/OnePunch.com';

        test('Should Match Snapshot',
            () => {
                const { container } = render(<GifItem title={testTitle} url={testUrl} />);
                expect(container).toMatchSnapshot();
            });

        test('Should have title and url props',
            () => {

                render(<GifItem title={testTitle} url={testUrl} />);

                // expect(screen.getByText(testTitle)).toBeTruthy();
                // expect(screen.getByRole('img').src).toBe(testUrl);
                // expect(screen.getByRole('img').alt).toBe(testUrl);

                // Or try this...

                const { src, alt } = screen.getByRole('img');
                expect(src).toBe(testUrl);
                expect(alt).toBe(testTitle);
            })

        test('Should render components title',
            () => {
                render(<GifItem title={testTitle} url={testUrl} />);

                expect(screen.getByText(testTitle)).toBeTruthy();

            })
    });