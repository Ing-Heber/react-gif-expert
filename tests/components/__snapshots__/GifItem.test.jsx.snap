// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Component Should Match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://localhost/OnePunch.com"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
